src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -104,214 +109,196 @@ private void tagIde(String ideLabel, String version) {
     }
 
     private void captureCiOrLocal() {
-        buildScan.tag(isCi() ? "CI" : "LOCAL");
+        buildScan.tag(CiUtils.isCi(providers) ? "CI" : "LOCAL");
     }
 
     private void captureCiMetadata() {
-        if (isJenkins() || isHudson()) {
-            Optional<String> buildUrl = envVariable("BUILD_URL");
-            Optional<String> buildNumber = envVariable("BUILD_NUMBER");
-            Optional<String> nodeName = envVariable("NODE_NAME");
-            Optional<String> jobName = envVariable("JOB_NAME");
-            Optional<String> stageName = envVariable("STAGE_NAME");
-
-            buildUrl.ifPresent(url ->
-                buildScan.link(isJenkins() ? "Jenkins build" : "Hudson build", url));
-            buildNumber.ifPresent(value ->
-                buildScan.value("CI build number", value));
-            nodeName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI node", value));
-            jobName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI job", value));
-            stageName.ifPresent(value ->
-                customValueSearchLinker.addCustomValueAndSearchLink("CI stage", value));
-
-            jobName.ifPresent(j -> buildNumber.ifPresent(b -> {
-                Map<String, String> params = new LinkedHashMap<>();
-                params.put("CI job", j);
-                params.put("CI build number", b);
-                customValueSearchLinker.registerLink("CI pipeline", params);
-            }));
-        }
+        gradle.afterProject(project -> {
```

Comment:
is this `afterProject` still needed given the entire captureCiMetadata is performed inside `buildFinished` ?

Reply:
Yes, if not, you're getting `root project is not yet available for build` when passing the the project directory parameter to the `Action`